app/parse.py

```diff
@@ -14,5 +19,30 @@ class Course:
     type: CourseType
 
 
+def parse_single_course(product_soup: BeautifulSoup, type_: str) -> Course:
+    return Course(
+        name=product_soup.select_one("span.typography_landingH3__vTjok").text,
+        short_description=product_soup.select_one("p").text,
+        type=CourseType(type_)
+    )
+
+
 def get_all_courses() -> list[Course]:
-    pass
+    page = requests.get(BASE_URL).content
+    soup = BeautifulSoup(page, "html.parser")
+    all_courses = {
+        type_.value: soup.select(
+            f"div[id={type_.value}] section.CourseCard_cardContainer__7_4lK"
+        )
+        for type_ in CourseType
+    }
+    courses_list = []
+
+    for key, value in all_courses.items():
+        for data in value:
+            courses_list.append(parse_single_course(data, key))
+    return courses_list
```

Comment:
you can simplify this block, just combine everything in one cycle. Iterate through course types, select courses, and call `parse_single_course` with soup and course type. Creating a dictionary is unnecessary.